src/sentry/api/base.py

```diff
@@ -497,14 +498,16 @@ def resolve_region(request: Request):
 class EndpointSiloLimit(SiloLimit):
     def modify_endpoint_class(self, decorated_class: Type[Endpoint]) -> type:
         dispatch_override = self.create_override(decorated_class.dispatch)
-        return type(
+        new_class = type(
             decorated_class.__name__,
             (decorated_class,),
             {
                 "dispatch": dispatch_override,
                 "__silo_limit": self,  # For internal tooling only
             },
         )
+        new_class.__module__ = decorated_class.__module__
```

Comment:
This fixes a pretty gnarly bug where the decorator classes were showing up with the module `sentry.api.base` in the `audit_silo_decorators` script, which messed up repeated runs.

Reply:
AFAIK, functools should offer a way to avoid this from happening. Not sure if it works for class-based decorators though.

Reply:
Yes, unfortunately `functools.update_wrapper` works only for functions. I don't know of an equivalent tool for whole classes.